codegen/src/types/composite_def.rs

```diff
@@ -210,11 +259,38 @@ impl CompositeDefFields {
         }
     }
 
+    /// Generate the code for type aliases which will be used to construct the `struct` or `enum`.
+    pub fn to_type_aliases_tokens(&self, visibility: Option<&syn::Visibility>) -> TokenStream {
```

Comment:
I still feel uneasy about this and the below `alias_module_name` additions (which, if provided, basically falls into if branches which do something that takes up most of the logic there).

I wonder whether this alias stuff should be more completely separate from the usual type generation stuff in this `types` folder, so that the `types` folder is relatively untouched in this PR (and can be swapped out for `scale-typegen` fairly easily to do the standard type generation stuff), and the alias stuff lives elsewhere (maybe a separate `generate_alias.rs` file or something outside of this folder) and is quite independent?

I'm not sure whether that would lead to a bunch of code duplication though, or whether the code for aliases is different enough that there's actually not a lot of overlap (which is just the feeling I get looking at the below).

That all said, @tadeohepperle what are your thoughts on this? Does adding these bits complicate moving to `scale-typegen`? We can also think about it in a separate PR of course too, since I know that this one has been lingering a while!

Reply:
I think it will be fine to move the `to_type_alias_tokens` into a standalone function later when integrating `scale-typegen` (because `CompositeDefFields` will no longer exist). Let's merge this and I will figure out the conflicts things out in this PR: https://github.com/paritytech/subxt/pull/1260